lib/atlas/scaler/graph_scaler.rb

```diff
@@ -0,0 +1,40 @@
+module Atlas
+  class Scaler::GraphScaler
+    # Edge attributes which must be scaled.
+    EDGE_ATTRIBUTES = [:demand].freeze
+
+    # Node attributes which must be scaled.
+    NODE_ATTRIBUTES = [
+      :demand,
+      :max_demand,
+      :typical_input_capacity,
+      :electricity_output_capacity
+    ].freeze
+
+    # Maps top-level keys from the dumped graph to arrays of attributes which
+    # need to be scaled.
+    SCALED_ATTRIBUTES = {
+      edges: EDGE_ATTRIBUTES,
+      nodes: NODE_ATTRIBUTES
+    }.freeze
+
+    def initialize(scaling_factor)
+      @scaling_factor = scaling_factor
+    end
+
+    # Public: Scales the demands in the graph - modifying the original graph!
```

Comment:
I think this documentation is out-of-date. It looks like `graph` is actually the hash of node and edge data, and it doesn't return the graph object anymore.